cypress/integration/collections_spec.js

```diff
@@ -0,0 +1,123 @@
+      cy.get('@collections').should('have.attr', 'href', '#/collections');
+      cy.get('@collections').click();
+
+      cy.url().should('include', 'collections');
+      cy.get('.heading--xlarge').should('have.text', 'Collections');
+
+      cy.get('table tbody tr').its('length').should('be.eq', 5);
+    });
+
+    it('collections page displays a button to add a new collection', () => {
+      cy.visit('/#/collections');
+
+      cy.get('.heading--large').should('have.text', 'Collection Overview');
+      cy.contains('a', 'Add a Collection').should('exist').as('addCollection');
+      cy.get('@addCollection').should('have.attr', 'href', '#/collections/add');
+      cy.get('@addCollection').click();
+
+      // fill the form and submit
+      const collection = '{{}"name":"TESTCOLLECTION","version":"006","dataType":"TESTCOLLECTION", "duplicateHandling":"error"}';
+      cy.get('textarea').type(collection, {force: true});
+      cy.get('form').get('input').contains('Submit').click();
+
+      // displays the new collection
+      cy.get('.heading--xlarge').should('have.text', 'Collections');
+      cy.get('.heading--large').should('have.text', `${name} / ${version}`);
+      cy.url().should('include', `#/collections/collection/${name}/${version}`);
+
+      // verify the new collection is added to the collections list
+      cy.contains('a', 'Back to Collections').click();
+      cy.contains('table tbody tr a', name)
+        .should('exist')
+        .and('have.attr', 'href', `#/collections/collection/${name}/${version}`);
+    });
+
+    it('collection page has button to edit the collection', () => {
+      cy.visit(`/#/collections/collection/${name}/${version}`);
+      cy.contains('a', 'Edit').should('exist').as('editCollection');
+      cy.get('@editCollection')
+        .should('have.attr', 'href')
+        .and('include', `#/collections/edit/${name}/${version}`);
+      cy.get('@editCollection').click();
+
+      cy.get('.heading--large').should('have.text', `Edit ${name}___${version}`);
+
+      // There are issues updating the react form which uses ace editor.
+      // Neither cy .clear nor ace editor works.
+      //
+      // cy .clear or .type('{selectall}{backspace}' has the following error:
+      // CypressError: cy.type() failed because this element is detached from the DOM.
+      //
+      // ace editor: editor.setValue(collection) does update the value in the ace_content,
+      // but doesn't cause the textarea value to be updated, and the updated conent is not submitted
+      // with the form.
+      // const collection = '{"name":"TESTCOLLECTION","version":"006","dataType":"TESTCOLLECTION", "duplicateHandling":"replace"}';
+      // const escapedCollection = collection.replace('{', '{{}');
+      // cy.get('textarea')
+      //   .type('{selectall}{backspace}', { force: true })
+      //   .type(escapedCollection, { force: true });
+      // cy.get('.ace_editor').then((ele) => {
```

Comment:
Even though this is commented out, I think we can remove it since I don't think this approach will ever work